webpack/assets/javascripts/react_app/components/HostDetails/index.js

```diff
@@ -56,12 +57,18 @@ const HostDetails = ({
   }, [tabs]);
 
   useEffect(() => {
-    registerCoreTabs();
-  }, []);
+    if (computeId) {
+      !isConsoleRegistered.current && registerConsoleTab();
+      isConsoleRegistered.current = true;
+    } else if (isConsoleRegistered.current) {
+      dispatch(unregisterTab('Console'));
+    }
+  }, [computeId]);
```

Comment:
This is for register the console tab dynamically.
 if a compute resource is assigned the console tab is registered, if not the console tab is removed  (only when it already registered)